cpp/src/arrow/util/value_parsing.h

```diff
@@ -281,6 +305,16 @@ struct StringToUnsignedIntConverterMixin {
     if (ARROW_PREDICT_FALSE(length == 0)) {
       return false;
     }
+    // If it starts with 0x then its hex
+    if (length > 2 && s[0] == '0' && ((s[1] == 'x') || (s[1] == 'X'))) {
+      length -= 2;
+      s += 2;
+
+      if (!ARROW_PREDICT_TRUE(ParseHex(s, length, out))) {
+        return false;
+      }
+      return true;
```

Comment:
Simplify to `return ARROW_PREDICT_TRUE(ParseHex(s, length, out));`.

Reply:
done